# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: netwerk/ipc/DocumentChannel.cpp
# Commit: d2fd0f955e24
# Full Hash: d2fd0f955e2462852c0a37ca73267b20ac0be9df
# Author: Dan Glastonbury <dan.glastonbury@gmail.com>
# Date: 2020-07-13 09:51:22
# Regressor Bug: 1646899
# File Overlap Count: 1
# Description:
#   Bug 1646899 - P3: Modify CanUseDocumentChannel to take necessary vars. r=mattwoodrow
#   
#   Allows checking if DocumentChannel can be used with out creating an
#   nsDocShellLoadState object.
#   
# ==============================================================================

diff -r 7a7194146fe1 -r d2fd0f955e24 netwerk/ipc/DocumentChannel.cpp
--- a/netwerk/ipc/DocumentChannel.cpp	Mon Jul 13 00:48:52 2020 +0000
+++ b/netwerk/ipc/DocumentChannel.cpp	Mon Jul 13 00:48:54 2020 +0000
@@ -154,9 +154,7 @@
          !spec.EqualsLiteral("about:crashcontent");
 }
 
-bool DocumentChannel::CanUseDocumentChannel(nsDocShellLoadState* aLoadState) {
-  MOZ_ASSERT(aLoadState);
-
+bool DocumentChannel::CanUseDocumentChannel(nsIURI* aURI, uint32_t aLoadFlags) {
   if (XRE_IsParentProcess() &&
       !StaticPrefs::browser_tabs_documentchannel_ppdc()) {
     return false;
@@ -168,8 +166,8 @@
   // outer document (and must load in the same process), which breaks if we
   // serialize to the parent process.
   return StaticPrefs::browser_tabs_documentchannel() &&
-         !aLoadState->HasLoadFlags(nsDocShell::INTERNAL_LOAD_FLAGS_IS_SRCDOC) &&
-         URIUsesDocChannel(aLoadState->URI());
+         !(aLoadFlags & nsDocShell::INTERNAL_LOAD_FLAGS_IS_SRCDOC) &&
+         URIUsesDocChannel(aURI);
 }
 
 /* static */